function-validation/src/main/java/com/future/function/validation/validator/core/FileMustBeImageValidator.java

```diff
@@ -38,12 +38,24 @@ public boolean isValid(
     return Optional.ofNullable(value)
       .orElseGet(Collections::emptyList)
       .stream()
-      .map(fileRepositoryV2::findOne)
-      .filter(Objects::nonNull)
+      .map(fileRepositoryV2::findByIdAndDeletedFalse)
+      .allMatch(this::hasRecognizedImageExtension);
+  }
+
+  private Boolean hasRecognizedImageExtension(Optional<FileV2> retrievedFile) {
+
+    return retrievedFile.map(this::getFileExtension)
+      .map(ext -> fileProperties.getImageExtensions()
```

Comment:
when the ext is null from method `getFileExtension`, what will happen?

Reply:
It will redirect to the `.orElse(Boolean.FALSE)` line, meaning the file has no extension.